docs/test_scripts/test_basic_checkpointing.py

```diff
@@ -0,0 +1,67 @@
+# [imports]
+import torch
+import numpy as np
+import modulus
+import matplotlib.pyplot as plt
+from modulus.datapipes.benchmarks.darcy import Darcy2D
+from modulus.models.fno.fno import FNO
+from modulus.metrics.general.mse import mse
+from modulus.launch.utils import load_checkpoint, save_checkpoint
+# [imports]
+
+# [code]
+normaliser = {
+    "permeability": (1.25, 0.75),
+    "darcy": (4.52e-2, 2.79e-2),
+}
+dataloader = modulus.datapipes.benchmarks.darcy.Darcy2D(
+    resolution=256, batch_size=64, nr_permeability_freq=5, normaliser=normaliser
+)
+model = FNO(
+    in_channels=1,
+    out_channels=1,
+    decoder_layers=1,
+    decoder_layer_size=32,
+    dimension=2,
+    latent_channels=32,
+    num_fno_layers=4,
+    num_fno_modes=12,
+    padding=5,
+).to("cuda")
+
+optimizer = torch.optim.Adam(model.parameters(), lr=0.01)
+scheduler = torch.optim.lr_scheduler.LambdaLR(
+    optimizer, lr_lambda=lambda step: 0.85**step
+)
+
+# load the epoch and optimizer, model ans scheduler parameters from the checkpoint if 
+# it exists 
+loaded_epoch = load_checkpoint(
+    "./checkpoints",
+    models=model,
+    optimizer=optimizer,
+    scheduler=scheduler,
+    device="cuda",
+)
+
+# run for 20 iterations
+# starting with the loaded epoch
+for i in range(max(1, loaded_epoch), 21):
```

Comment:
Make distinction between epoch and iteration

Reply:
Done!